File: Beheersfuncionaliteiten.java
package com.example.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/api")
public class MainController {

    @GetMapping("/data")
    public Map<String, String> fetchDataFromServer() {
        // Simuleer het ophalen van gegevens van de server
        Map<String, String> responseData = new HashMap<>();
        responseData.put("message", "Dit is een serverreactie!");
        return responseData;
    }
}

File: Beheersfuncionaliteiten.js
document.addEventListener('DOMContentLoaded', function () {
    // Simuleer het sturen van een HTTP-verzoek naar de server
    fetch('/api/data') // Stel de juiste URL in voor je server
        .then(response => response.json())
        .then(data => {
            // Toon de serverreactie op de pagina
            const responseElement = document.getElementById('server-response');
            responseElement.textContent = Serverreactie: ${data.message};
        })
        .catch(error => console.error('Fout bij het ophalen van servergegevens:', error));
});
